### DISCUSSION TITLE
Getting "context deadline exceeded" error

### DISCUSSION THREAD

**Discussion by Praveen5k** [prometheus/prometheus#10491] [Q&A]
We are getting the below error intermittently in Prometheus and whenever this error occurs, the particular target "up" value is showing as 0 which results in triggering a false alert. I have gone through some forums and they have suggested to increase scrape_timeout value. I have set scrape_interval as 60s, so I cannot increase scrape_timeout beyond 60s. Currently, scrape_timeout value is set as 10s by default. I am getting the below error for some other targets as well. But after 2 or 3 minutes, Prometheus is able to fetch metrics and shows target "up" value as 1. Can someone help me to understand why this issue is occurring and resolve this issue? 

**Error:**
Mar 20 06:14:42 promethhost prometheus: level=debug ts=2022-03-20T10:14:42.764Z caller=scrape.go:1137 component="scrape manager" scrape_pool=zookeeper target=[http://xxx.zookeeper.xxx:9100/metrics](https://urldefense.proofpoint.com/v2/url?u=http-3A__xxx.zookeeper.xxx-3A9100_metrics&d=DwQF_g&c=jf_iaSHvJObTbx-siA1ZOg&r=QmRSslm4rpGcu4fLxdJhB1LBDgyF5D-MrdvR4kv4AZs&m=KJeNnRpwh0lLgpLERqGfSp7yvXxrAnNmMvj8Rl4CERE&s=6y3vmtXXKKWA_cl_z7kJlQcEQexbdcuS8IapADOIxtA&e=) msg="Scrape failed" err="Get \[http://xxx.zookeeper.xxx:9100/metrics\](https://urldefense.proofpoint.com/v2/url?u=http-3A__xxx.zookeeper.xxx-3A9100_metrics_&d=DwMF_g&c=jf_iaSHvJObTbx-siA1ZOg&r=QmRSslm4rpGcu4fLxdJhB1LBDgyF5D-MrdvR4kv4AZs&m=KJeNnRpwh0lLgpLERqGfSp7yvXxrAnNmMvj8Rl4CERE&s=ujjCloGYQnpdA9xjFjmhgkF0oh_CE0DPSbwKoYld-EI&e=): context deadline exceeded"

**Comment by LeviHarrison** [comment]
This errors means that the scrape request is timing out because the target isn't responding fast enough. There's probably something wrong with that ZooKeeper instance.

**Reply by Praveen5k** [reply]
We are getting this error intermittently. We checked the network-related stuff on zookeeper servers. But there were no errors found from the network side and everything looks normal. Will there be any chance for any other issue from Prometheus end?

**Reply by LeviHarrison** [reply]
Given that it's intermittent, no, I really don't think so. You could try increasing the timeout further if you also increase the interval.
